packages/experimental/NativeFontMetrics/ios/FRNFontMetrics.h

```diff
@@ -0,0 +1,31 @@
+#import <React/RCTBridgeModule.h>
+#import <React/RCTEventEmitter.h>
+
+@interface FRNFontMetrics : RCTEventEmitter <RCTBridgeModule>
+@end
+
+#import <React/RCTConvert.h>
+
+typedef NS_ENUM(NSInteger, FRNTextStyle) {
+    FRNTextStyleUndefined,
+    FRNTextStyleCaption2,
+    FRNTextStyleCaption1,
+    FRNTextStyleFootnote,
+    FRNTextStyleSubheadline,
+    FRNTextStyleCallout,
+    FRNTextStyleBody,
+    FRNTextStyleHeadline,
+    FRNTextStyleTitle3,
+    FRNTextStyleTitle2,
+    FRNTextStyleTitle1,
+    FRNTextStyleLargeTitle
+};
+
+@interface RCTConvert (FRNFontMetrics)
+
++ (FRNTextStyle)FRNTextStyle:(nullable id)json;
+
+@end
+
+UIFontMetrics *_Nonnull FRNUIFontMetricsForTextStyle(FRNTextStyle textStyle);
+CGFloat FRNBaseSizeForTextStyle(FRNTextStyle textStyle);
```

Comment:
Any reason these two can't/shouldn't be class methods on `RCTConvert`? Why C functions? Sorry if there's a good reason that I don't have familiarity with.

Reply:
Looking at the `.m` file, I think it's so that we can do this?

```
  FRNTextStyle style = [RCTConvert FRNTextStyle:styleString];
  UIFontMetrics *fontMetrics = FRNUIFontMetricsForTextStyle(style);
```

Reply:
`RCTConvert` is specifically for deserialization of an `id` to a native value. It's not really meant to do any other logic that these C functions are doing.